vsintegration/src/FSharp.Editor/QuickInfo/QuickInfoProvider.fs

```diff
@@ -156,101 +157,125 @@ module private FSharpQuickInfo =
+            | _ ->
                 let! symbolUse = checkFileResults.GetSymbolUseAtLocation (textLineNumber, symbol.Ident.idRange.EndColumn, textLine.ToString(), symbol.FullIsland, userOpName=FSharpQuickInfo.userOpName)
                 let! symbolSpan = RoslynHelpers.TryFSharpRangeToTextSpan (sourceText, symbol.Range)
                 return res, symbolSpan, symbolUse.Symbol, symbol.Kind
         }
-    
-    interface IQuickInfoProvider with
-        override this.GetItemAsync(document: Document, position: int, cancellationToken: CancellationToken): Task<QuickInfoItem> =
-            asyncMaybe {
-                let! symbolUse, sigTooltipInfo, targetTooltipInfo = 
-                    FSharpQuickInfo.getTooltipInfo(checkerProvider.Checker, projectInfoManager, document, position, cancellationToken)
 
-                match sigTooltipInfo, targetTooltipInfo with 
-                | None, None -> return null
-                | Some tooltip, None  
-                | None, Some tooltip -> 
-                    let mainDescription, documentation, typeParameterMap, usage, exceptions = ResizeArray(), ResizeArray(), ResizeArray(), ResizeArray(), ResizeArray()
-                    XmlDocumentation.BuildDataTipText(documentationBuilder, mainDescription.Add, documentation.Add, typeParameterMap.Add, usage.Add, exceptions.Add, tooltip.StructuredText)
-                    let glyph = Tokenizer.GetGlyphForSymbol(tooltip.Symbol, tooltip.SymbolKind)
-                    let navigation = QuickInfoNavigation(gotoDefinitionService, document, symbolUse.RangeAlternate)
-                    let content = viewProvider.ProvideContent(glyph, mainDescription, documentation=documentation, typeParameterMap=typeParameterMap, usage=usage, exceptions=exceptions, navigation=navigation)
-                    return QuickInfoItem (tooltip.Span, content)
+    interface IAsyncQuickInfoSource with
+        override __.Dispose() = () // no cleanup necessary
+        override __.GetQuickInfoItemAsync(session:IAsyncQuickInfoSession, cancellationToken:CancellationToken) : Task<QuickInfoItem> =
+            let triggerPoint = session.GetTriggerPoint(textBuffer.CurrentSnapshot)
+            match triggerPoint.HasValue with
+            | false -> Task.FromResult<QuickInfoItem>(null)
+            | true ->
+                let triggerPoint = triggerPoint.GetValueOrDefault()
+                let xmlMemberIndexService = serviceProvider.GetService(typeof<SVsXMLMemberIndexService>) :?> IVsXMLMemberIndexService
+                let documentationBuilder = XmlDocumentation.CreateDocumentationBuilder(xmlMemberIndexService, serviceProvider.DTE)
+                asyncMaybe {
+                    let document = textBuffer.CurrentSnapshot.GetOpenDocumentInCurrentContextWithChanges()
+                    let! symbolUse, sigTooltipInfo, targetTooltipInfo = FSharpQuickInfo.getTooltipInfo(checkerProvider.Checker, projectInfoManager, document, triggerPoint.Position, cancellationToken)
```

Comment:
tooltips and quickinfo are different apis, could you rename these to quickinfo here

Reply:
Thanks, I've added a commit that does the appropriate renames.